bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -101,6 +107,68 @@ public MessageBox (Shell parent, int style) {
 	checkSubclass ();
 }
 
+long CBTProc(long nCode, long wParam, long lParam) {
+	if (labels != null && (int) nCode == OS.HCBT_ACTIVATE) {
+		/* The system is about to activate MessageBox window */
+		switch (buttonBits) {
+		case OS.MB_OKCANCEL:
+			setButtonText(wParam, SWT.CANCEL, OS.IDCANCEL);
```

Comment:
Just a small rant, feel free to ignore: I would rather not use case fallthrough because it makes code less readable.

Also note that instead of `buttonBits` it was wiser to just use `this.style` and avoid `switch` completely.